test/util/test_parse.py

```diff
@@ -40,6 +43,79 @@ def test_extractnumber(self):
         self.assertEqual(extractnumber("three quarter cups"), 3.0/4.0)
         self.assertEqual(extractnumber("three quarters cups"), 3.0/4.0)
 
+    def test_extractdatetime_en(self):
+
+        def extractWithFormat(text):
+            date = datetime(2017, 06, 27, 00, 00)
+            [extractedDate, leftover] = extract_datetime(text, date)
+            extractedDate = extractedDate.strftime("%Y-%m-%d %H:%M:%S")
+            return [extractedDate, leftover]
+
+        self.assertEqual(extractWithFormat("Set the ambush for 5 days from today")[0], "2017-07-02 00:00:00")  # noqa
+        self.assertEqual(extractWithFormat("Set the ambush for 5 days from today")[1], "set ambush")  # noqa
```

Comment:
I don't want to be pedantic, but you could combine these into a single test function, e.g. something like:

```
def testExtract(text, expected_date, expected_leftover):
    res = extractWithFormat(text)
    self.assertEqual(res[0], expected_date)
    self.assertEqual(res[1], expected_leftover)

testExtract("Set the ambush for 5 days from today",
                "2017-07-02 00:00:00", "set ambush")
testExtract("What is the day after tomorrow's weather",
                "2017-06-29 00:00:00", "what is weather")
```

It'll be much more compact, less typing and more obvious.